### DISCUSSION TITLE
⏳ [Subscription]: Returning an array of entities that match with filter criteria from @Args for Subscription resolvers

### DISCUSSION THREAD

**Discussion by Mingyang-Li** [MichalLytek/type-graphql#1343] [Q&A]
Hello friends, I'm looking for a way to implement `Subscription` resolvers that return an array of entiries which match the filter criteria from `@Args()` passed in

I haven't found any issues nor PR related to this feature.

Originally submitted a similar same issue (with LOTS of details) for NestJS implementation in [their repo](https://github.com/nestjs/graphql/issues/2385), but sadly they're not implementing it in the forseable future.

Yet, I need to figure out a way to do this.

Any help would be appreciated:)

**Comment by MichalLytek** [comment]
`@Subscription` decorator has a `filter` argument where you can pass your function that filters payloads from pubsub and decide if you want to trigger response to client from subscription or not for that event.

Is that something you're describing? I'm not sure what kind of filtering you want.

**Reply by Mingyang-Li** [reply]
hi @MichalLytek, I was aware of the `filter` argument from `@Subscription` decorator. However, I doubt it if this feature alone can help me build a subscription resolver that allows me to do something like this:
```gql
subscription {
  subscribeUsers (
    where: {
      id: { _eq: 'djfdf2387b' },
      createdAt: { _gte: '2022-08-31' }
    },
    orderBy: {
      id: Asc,
      amount: Desc
      createdAt: Asc
    },
    skip: 4,
    take: 6,
  ) {
     id
     firstName
     paymentsCreated {
       id
       amount
       currency
     }
  }
}
```

Which, returns me an array of `User` entities that match the filters from arguments within `subscribeUsers`

**Reply by MichalLytek** [reply]
`filter` allows preventing triggering subscription on pubsub event. You can also do `Array.prototype.filter` in your resolver to filter the returned array.

Based on your example in NestJS repo, you should setup pubsub to trigger topic when something changes in observed table and then in the subscription resolver, just fetch the data from db using args as sql where conditions. Does it solves your problem?

**Reply by Mingyang-Li** [reply]
Hmmm, so did you mean I need to query the updated entities on `create` and `update` instead of within `subsciption` resolver?

For e.g,:
```ts
import { Args, Mutation, Query, Resolver, Subscription } from '@nestjs/graphql';
import { PubSub } from 'graphql-subscriptions';
import { CreatePaymentArgs } from 'src/payment/dtos/CreatePaymentArgs';
import { UpdatePaymentArgs } from 'src/payment/dtos/UpdatePaymentArgs';
import { Payment } from './dtos/Payment';
import { PaymentFindManyArgs } from './dtos/PaymentFindManyArgs';
import { PaymentService } from './payment.service';

@Resolver(() => Payment)
export class PaymentResolver {
  constructor(private paymentService: PaymentService, private pubSub: PubSub) {}

  @Mutation(() => Payment)
  public async createPayment(
    @Args() args: CreatePaymentArgs,
  ): Promise<Payment> {
    const newPayment = await this.paymentService.create(args);
    if (newPayment) {
      const payments = this.paymentService.findMany(args);
      this.pubSub.publish('paymentsUpdated', payments);
    };
    return newPayment;
  }

  @Mutation(() => Payment)
  public async updatePayment(
    @Args() args: UpdatePaymentArgs,
  ): Promise<Payment> {
    const updatedPayment = await this.paymentService.update(args);
    if (updatedPayment) {
      const payments = this.paymentService.findMany(args);
      this.pubSub.publish('paymentsUpdated', payments);
    };
    return updatedPayment;
  }

  @Query(() => [Payment])
  public async payments(@Args() args: PaymentFindManyArgs): Promise<Payment[]> {
    return this.paymentService.findMany(args);
  }

  @Subscription(() => [Payment])
  public async subscribePayments(@Args() args: PaymentFindManyArgs) {
    return this.pubSub.asyncIterator('paymentsUpdated');
  }
}
```

I updated my `updatePayment` and `createPayment` resolvers, to get `PubSub` to publish an array of `Payments` on update or create, do you think that's a feasible solution?
